www/js/factory/address.factory.js

```diff
@@ -17,5 +17,23 @@ angular.module('vamosJuntas').factory('addressFactory', function($http, PlacesAP
       url = PlacesAPI.nearbySearchBaseUrl + "location=" + lat + "," + long + "&radius=300&language=pt-PT&key=" + api_key;
       return $http.get(url);
     },
+
+    getAddressByCoord: function(latitude, longitude) {
+      var deferred = $q.defer();
+      var geocoder = new google.maps.Geocoder();
+      var latlng = {
+          lat: latitude,
+          lng: longitude
+        };
+
+      geocoder.geocode({'location': latlng}, function(results, status) {
+        if (status === google.maps.GeocoderStatus.OK && results.length > 0) {
+          deferred.resolve(results[0].formatted_address);
+        } else {
+          deferred.reject(status);
+        }
+      });
+      return deferred.promise;
+    }
```

Comment:
we should be using the response of the request in the same factory, no need to send it back to the controller's scope.